### PR TITLE
Bugfix/3860 transaction picker

### PR DESCRIPTION
**Pull request by MarekM25**
#### Changes
We won't be able to pick invalid 3860 transactions for block production.

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [x] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs [hunk 1]
```diff
@@ -40,14 +40,11 @@ public bool IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec)
                    ValidateChainId(transaction) &&
                    Validate1559GasFields(transaction, releaseSpec) &&
                    Validate3860Rules(transaction, releaseSpec) &&
+                   Validate3860Rules(transaction, releaseSpec) &&
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs:43]: duplicate?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs (hunk 1 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs]: thx

**Review by damian-orzechowski — APPROVED** [review]
All looks good now.

**Review by marcindsobczak — APPROVED** [review]

File: src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.BlockProductionTransactionPicker.cs [hunk 2]
```diff
@@ -55,6 +55,11 @@ public AddingTxEventArgs CanAddTransaction(Block block, Transaction currentTx, I
                 }
 
                 IReleaseSpec spec = _specProvider.GetSpec(block.Header);
+                if (currentTx.IsAboveInitCode(spec))
+                {
+                    return args.Set(TxAction.Skip, $"EIP-3860 - transaction size over max init code size");
+                }
+
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.BlockProductionTransactionPicker.cs:62]: Question - can we do something that we won't need to duplicate this logic here?

It is partly done by TxPool, but it is not enough.

Comment (MarekM25) [reply src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.BlockProductionTransactionPicker.cs:62]: I think this is the case for every condition. We're duplicating logic in BlockProcessor.BlockProductionTransactionPicker, TxPool, TransactionExecutor. We should refactor everything somehow. I planned to create the issue

**Pull request merged**